### PR TITLE
Fix norm 1 crash in PANOC

### PR DESCRIPTION
**Pull request by korken89**
If the Lipschitz constant was estimated to 0 at init, the solver would
get NaN as gamma, causing failure. Now gamma is estimated using the
current cost and gradient, which in limited experiments seems to be an
acceptable heuristic.

The heuristic is only applied if  the Lipschitz constant was estimated to 0.

### CODE REVIEW

**Comment by alphaville** [comment]
Thank you, @korken89. I'll have a look.

File: examples/abs_nan.rs [hunk 1]
```diff
@@ -0,0 +1,37 @@
+/// Example for testing performance in problems PANOC was not made for.
```
Comment (alphaville) [inline examples/abs_nan.rs:1]: We should write a unit test.

**Review by alphaville — COMMENTED** [review]
We're getting a compilation error at https://travis-ci.org/github/alphaville/optimization-engine/jobs/757519401 - I'll pull this and have a look.

**Comment by alphaville** [comment]
Clippy gives some minor warnings, but returns an error. However, `cargo build` and `cargo test` work just fine.

**Comment by alphaville** [comment]
Clippy errors are now fixed. New error: [`RuntimeError: Python version >= 3.7 required`](https://travis-ci.org/github/alphaville/optimization-engine/jobs/757541344#L1031).

File: examples/abs_nan.rs (hunk 1 above)
Comment (korken89) [reply examples/abs_nan.rs:1]: I'll fix!

**Comment by alphaville** [comment]
There's a strange issue on [Trusty](https://travis-ci.org/github/alphaville/optimization-engine/jobs/757558374): `Couldn't find index page for 'retry'`. It works on Xenial and OSX, though. Since we'll be moving to GA soon, I'll remove Trusty from Travis so that we can merge this.

**Pull request closed without merging**